torchvision/prototype/datasets/utils/_resource.py

```diff
@@ -91,31 +91,30 @@ def load(
     ) -> IterDataPipe[Tuple[str, IO]]:
         root = pathlib.Path(root)
         path = root / self.file_name
+
         # Instead of the raw file, there might also be files with fewer suffixes after decompression or directories
         # with no suffixes at all.
         stem = path.name.replace("".join(path.suffixes), "")
-
-        # In a first step, we check for a folder with the same stem as the raw file. If it exists, we use it since
-        # extracted files give the best I/O performance. Note that OnlineResource._extract() makes sure that an archive
-        # is always extracted in a folder with the corresponding file name.
+        # Although it looks like we could glob for f"{stem}*" to find the file candidates as well as the folder
+        # candidate simultaneously, that would also pick up other files that share the same prefix. For example, the
+        # test split of the stanford-cars dataset uses the files
+        # - cars_test.tgz
+        # - cars_test_annos_withlabels.mat
+        # Globbing for `"cars_test*"` picks up both.
```

Comment:
Is this comment needed? It feels a bit like "in case you had a bad idea, here's why it's a bad idea"

Also, would our tests fail if we were to use `f"{stem}*"`? If yes I think it's definitely safe to remove the comment

Reply:
I'll check after #6010 is merged. IIRC, the test only failed sometimes, since `pathlib.Path().glob()` is non-deterministic and sometimes returned the correct path first, but not always.

Reply:
I had not realized this comment was also about the `folder_candidate` logic and not just about the line below. I agree it's worth keeping it now, thanks for writing it